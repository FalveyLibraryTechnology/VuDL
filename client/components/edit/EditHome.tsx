import React from "react";
import ChildList from "./ChildList";
import Link from "next/link";

const EditHome = () => {
    return (
        <div>
            <h1>Editor</h1>
            <h2>Tools</h2>
            <ul>
                <li>
                    <Link href="/edit/newChild">Create New Top-Level Object</Link>
                </li>
            </ul>
            <h2>Contents</h2>
            <ChildList />
        </div>
    );
};

export default EditHome;
