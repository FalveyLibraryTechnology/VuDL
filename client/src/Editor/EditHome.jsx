import React from "react";
import ChildList from "./ChildList";

const EditHome = () => {
    return (
        <div>
            <p>Welcome to the editor!</p>
            <ChildList />
        </div>
    );
};

export default EditHome;
