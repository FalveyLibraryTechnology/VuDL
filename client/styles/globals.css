:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
