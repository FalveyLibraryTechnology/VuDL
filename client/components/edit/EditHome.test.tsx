import React from "react";
import { describe, expect, it } from "@jest/globals";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import EditHome from "./EditHome";

const mockUseEditorContext = jest.fn();
jest.mock("../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));

jest.mock("./children/ChildList", () => () => "ChildList");
jest.mock("./EditorSnackbar", () => () => "EditorSnackbar");
jest.mock("./parents/ParentsModal", () => () => "ParentsModal");
jest.mock("./StateModal", () => () => "StateModal");

describe("EditHome", () => {
    let editorValues;
    beforeEach(() => {
        editorValues = {
            action: {
                initializeCatalog: jest.fn(),
            },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
    });
    it("renders", () => {
        const wrapper = mount(<EditHome />);
        expect(editorValues.action.initializeCatalog).toHaveBeenCalled();
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
