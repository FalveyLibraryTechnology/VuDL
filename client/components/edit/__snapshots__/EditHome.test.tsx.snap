// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditHome renders 1`] = `
<div>
  <h1>
    Editor
  </h1>
  <h2>
    Tools
  </h2>
  <ul>
    <li>
      <Link
        href="/edit/newChild"
      >
        Create New Top-Level Object
      </Link>
    </li>
  </ul>
  <h2>
    Contents
  </h2>
  <ChildList />
</div>
`;
